### PR TITLE
add _as_mpl_axes method to HighLevelWCSWrapper

### LINKED ISSUE #10117: HighLevelWCSWrapper should implement a _as_mpl_axes method
WCSAxes objects accept high or low level WCS objects, but because the `HightLevelWCSWrapper` class dosen't implement an `_as_mpl_axes` method it can not be passed to the `projection=` kwarg in various places in the matplotlib API.

### PR DESCRIPTION
**Pull request by broulston**
### Description

This pull request is to add an _as_mpl_axes method to the HightLevelWCSWrapper class. 

Fixes #10117

### CODE REVIEW

**Comment by astrofrog** [comment]
@broulston - thank you! Could you add an entry under the new features section in the 4.1 part of the changelog?

**Comment by broulston** [comment]
@astrofrog I've added an entry to the changelog!

File: astropy/wcs/wcsapi/high_level_wcs_wrapper.py [hunk 1]
```diff
@@ -69,3 +69,10 @@ def axis_correlation_matrix(self):
         See `~astropy.wcs.wcsapi.BaseLowLevelWCS.axis_correlation_matrix`
         """
         return self.low_level_wcs.axis_correlation_matrix
+
+    @property
```
Comment (Cadair) [inline astropy/wcs/wcsapi/high_level_wcs_wrapper.py:73]: I think the best thing to do is to make this a method which calls the method on the low level object.
```suggestion
```

File: astropy/wcs/wcsapi/high_level_wcs_wrapper.py [hunk 2]
```diff
@@ -69,3 +69,10 @@ def axis_correlation_matrix(self):
...
+    def _as_mpl_axes(self):
+        """
+        See `~astropy.wcs.wcsapi.BaseLowLevelWCS.__as_mpl_axes`
+        """
+        return self.low_level_wcs._as_mpl_axes
```
Comment (Cadair) [inline astropy/wcs/wcsapi/high_level_wcs_wrapper.py:78]:
```suggestion
        return self.low_level_wcs._as_mpl_axes()
```

File: astropy/wcs/wcsapi/high_level_wcs_wrapper.py [hunk 3]
```diff
@@ -69,3 +69,10 @@ def axis_correlation_matrix(self):
         See `~astropy.wcs.wcsapi.BaseLowLevelWCS.axis_correlation_matrix`
         """
         return self.low_level_wcs.axis_correlation_matrix
+
+    @property
+    def _as_mpl_axes(self):
+        """
+        See `~astropy.wcs.wcsapi.BaseLowLevelWCS.__as_mpl_axes`
```
Comment (Cadair) [inline astropy/wcs/wcsapi/high_level_wcs_wrapper.py:76]:
```suggestion
        See `~astropy.wcs.wcsapi.BaseLowLevelWCS._as_mpl_axes`
```

**Comment by Cadair** [comment]
Thanks @broulston :smile:

**Pull request merged**